rclrs/src/node.rs

```diff
@@ -71,18 +65,30 @@ pub struct Node {
     pub(crate) subscriptions_mtx: Mutex<Vec<Weak<dyn SubscriptionBase>>>,
     time_source: TimeSource,
     parameter: ParameterInterface,
-    // Note: it's important to have those last since `drop` will be called in order of declaration
-    // in the struct and both `TimeSource` and `ParameterInterface` contain subscriptions /
-    // services that will fail to be dropped if the context or node is destroyed first.
-    pub(crate) rcl_node_mtx: Arc<Mutex<rcl_node_t>>,
-    pub(crate) rcl_context_mtx: Arc<Mutex<rcl_context_t>>,
+    pub(crate) handle: Arc<NodeHandle>,
+}
+
+/// This struct manages the lifetime of the rcl node, and accounts for its
+/// dependency on the lifetime of its context.
```

Comment:
Strictly speaking, the declaration order in the struct doesn't even matter because the implementation of the `Drop` trait for each handle is ensuring that the necessary `[...]_fini` function is being called on the dependent before the `Arc<[Dependency]Handle>` field will get naturally dropped after the `Drop` trait implementation has finished executing.

The key detail here is that the `Arc<[Dependency]Handle>` pattern ensures transitive dependencies will be kept alive for as long as each handle is alive. But I'll give each Handle struct some commentary to explain their purpose.

Reply:
I've updated the documentation and added a reference to Rust destructors: [ae6d5b8](https://github.com/ros2-rust/ros2_rust/pull/386/commits/ae6d5b895828e65dac89c0960d45430707a6a3a9)